homeassistant/components/eq3btsmart/climate.py

```diff
@@ -62,55 +62,35 @@ class Eq3Climate(Eq3Entity, ClimateEntity):
     _attr_precision = PRECISION_HALVES
     _attr_hvac_modes = list(HA_TO_EQ_HVAC.keys())
     _attr_preset_modes = list(Preset)
-    _attr_should_poll = False
-    _attr_available = False
     _attr_hvac_mode: HVACMode | None = None
     _attr_hvac_action: HVACAction | None = None
     _attr_preset_mode: str | None = None
     _target_temperature: float | None = None
 
     @callback
-    def _async_on_updated(self) -> None:
-        """Handle updated data from the thermostat."""
+    def _handle_coordinator_update(self) -> None:
+        """Handle updated data from the coordinator."""
 
         if self._thermostat.status is not None:
-            self._async_on_status_updated()
+            self._target_temperature = self._thermostat.status.target_temperature.value
+            self._attr_hvac_mode = EQ_TO_HA_HVAC[self._thermostat.status.operation_mode]
+            self._attr_current_temperature = self._get_current_temperature()
+            self._attr_target_temperature = self._get_target_temperature()
+            self._attr_preset_mode = self._get_current_preset_mode()
+            self._attr_hvac_action = self._get_current_hvac_action()
 
         if self._thermostat.device_data is not None:
```

Comment:
Shouldn't we guard this to verify if the firmware version changed? I don't expect serial number to change. If we guard it, it doesn't have to the lookup and update at every update (while the moments that this happens is way smaller)